build/source/engine/systemSolv.f90

```diff
@@ -263,9 +210,7 @@ subroutine systemSolv(&
  integer(i4b),parameter          :: ixBandMatrix=1002            ! named variable for the band diagonal matrix
  integer(i4b)                    :: ixSolve                      ! the type of matrix used to solve the linear system A.X=B
  integer(i4b),parameter          :: iJac1=1                      ! first layer of the Jacobian to print
- integer(i4b),parameter          :: iJac2=10                     ! last layer of the Jacobian to print
- !integer(i4b),parameter          :: iJac1=457                    ! first layer of the Jacobian to print
- !integer(i4b),parameter          :: iJac2=466                    ! last layer of the Jacobian to print
+ integer(i4b),parameter          :: iJac2=5                     ! last layer of the Jacobian to print
```

Comment:
In this case these are used for debugging, so I think that it is fine.

The other hard-coded values are more questionable, e.g., 
```fortran
 real(dp),parameter              :: fScaleLiq=0.01_dp            ! func eval: characteristic scale for volumetric liquid water content (-)
 real(dp),parameter              :: fScaleMat=10._dp             ! func eval: characteristic scale for matric head (m)
 real(dp),parameter              :: fScaleNrg=1000000._dp          ! func eval: characteristic scale for energy (J m-3)
 real(dp),parameter              :: xScaleLiq=0.1_dp             ! state var: characteristic scale for volumetric liquid water content (-)
 real(dp),parameter              :: xScaleMat=10._dp             ! state var: characteristic scale for matric head (m)
 real(dp),parameter              :: xScaleTemp=1._dp             ! state var: characteristic scale for temperature (K)
```
There are two ways to go here:
(1) Allow some parameters to be hard-coded. This can be done for algorithmic control parameters (such as these) where we do not want to give the user the opportunity to mess things up.
(2) Require all parameters to be visible, no matter what.

As noted in comments for